ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -33,8 +44,14 @@ cpdef traceback_to_frames(traceback, max_nframes):
     while tb is not None:
         if nframes < max_nframes:
             frame = tb.tb_frame
-            code = frame.f_code
-            lineno = 0 if frame.f_lineno is None else frame.f_lineno
+            # Python 3.11 moved PyFrameObject to internal C API and cannot be directly accessed from tstate
+            IF PY_MAJOR_VERSION >= 3 and PY_MINOR_VERSION >= 11:
+                code = PyFrame_GetCode(frame)
+                lineno = PyFrame_GetLineNumber(frame)
+                lineno = 0 if lineno is None else lineno
+            ELSE:
+                code = frame.f_code
+                lineno = 0 if frame.f_lineno is None else frame.f_lineno
```

Comment:
@P403n1x87 I was unsure if this is also required to be changed, since this is taking a frame directly from `PyException_GetTraceback()`. I couldn't find in documentation if there's an accessor function to get a frame from a traceback - what do you think?

Reply:
If there is no documented API to do this then I guess this is the best we can do. Frame objects are guaranteed to be there with Python < 3.11, so we should be safe.